`azurerm_log_analytics_data_export_rule` - support for the `destination_metadata` property (#17019)
hashicorp/terraform-provider-azurerm: website/docs/r/log_analytics_data_export_rule.html.markdown

* `enabled` - (Optional) Is this Log Analytics Data Export Rule enabled? Possible values include `true` or `false`. Defaults to `false`.

* `metadata` - (Optional) A `metadata` block as defined below.

---

> The `metadata` block exports the following attributes:

Reviewer: ```suggestion
The `metadata` block supports the following attributes:
```
Should this be `supports` instead of `exports`?

Author: Thanks. Updated

> The `metadata` block supports the following attributes:

Reviewer 2: is this metadata? or destinations? should be bother putting this inside a block?

Author: Updated to `destination_metadata`.

> The `metadata` block supports the following attributes:

Reviewer 2: is this metadata? or destinations? should be bother putting this inside a block?

Author: Closed this comment as duplicate

> * `eventhub_name` - (Optional) The name of the EventHub that is defined for the Log Analytics Data Export Rule.

Reviewer 2: as above, what is this actually doing? if this is setting anything? just a place to put the destination event hub name? should it just be
```suggestion
* `destination_eventhub_name` - (Optional) The name of the EventHub that Log Analytics Data Export Rule exports too.
```

also given we are giving the id above? what is the point of this? could we not just grab it from the ID and set it ourselves so the user doesn't have to?

Author: This new introduced property means "Select an Event Hub name to send data from all tables in the rule to a specific Event Hub. If nothing is selected, an Event Hub is created for each table in the selected namespace".

When this is setting anything, it would fail.

I think it's better to update `metadata` to `destination_metadata` and keep `eventhub_name` since maybe service team would introduce other destinations.

Reviewer 2: Well i think that's pretty self explanatory, we should match what the portal is doing. and really. going by these images,  maybe have two resources 1 for storage account, 1 for eventhubs - there no metadata in those pages, and the property names are much easier to decipher.

Author: Azure portal also allows to set or not set event hub name under destination metadata. So I assume my PR is aligned with Azure Portal. If I am wrong, please correct me.

Currently the destination metadata only supports event hub name.

I already updated the property name and the name description. Seems you're still referencing old commit. Could you refresh your PR to get my latest update?Thanks.

And suggest go thourgh this azure doc (https://docs.microsoft.com/en-us/azure/azure-monitor/logs/logs-data-export?tabs=portal#create-or-update-data-export-rule) and play this log analytics data export rule resource on Azure Portal first.

Or could you provide the suggested code/commit for them? Thanks in advance.

![image](https://user-images.githubusercontent.com/19754191/176070309-33697f11-0152-4551-b253-a21f1e5d63ef.png)
![image](https://user-images.githubusercontent.com/19754191/176070562-c9b0114c-fc54-4b8c-b9c1-90b03454613d.png)

Reviewer 2: What i'm thinking is we model the two twps you can select in the portal:
```
resource "azurerm_log_analytics_data_export_rule_storage_account" "test" {
  name                    = "acctest-DER-%d"
  resource_group_name     = azurerm_resource_group.test.name
  workspace_resource_id   = azurerm_log_analytics_workspace.test.id
  table_names             = ["Heartbeat"]

  destination_storage_account_id = 
}

resource "azurerm_log_analytics_data_export_rule_evenhub" "test" {
  name                    = "acctest-DER-%d"
  resource_group_name     = azurerm_resource_group.test.name
  workspace_resource_id   = azurerm_log_analytics_workspace.test.id
  table_names             = ["Heartbeat"]

  destination_evenhub_id = azurerm_eventhub.test.id
}
```
or even
```
resource "azurerm_log_analytics_data_export_rule" "test" {
  name                    = "acctest-DER-%d"
  resource_group_name     = azurerm_resource_group.test.name
  workspace_resource_id   = azurerm_log_analytics_workspace.test.id
  table_names             = ["Heartbeat"]

  destination_evenhub_id = azurerm_eventhub.test.id
  destination_storage_account_id = 
}
```

> * `eventhub_name` - (Optional) The name of the EventHub that is defined for the Log Analytics Data Export Rule.

Reviewer 2: as above, what is this actually doing? if this is setting anything? just a place to put the destination event hub name? should it just be
```suggestion
* `destination_eventhub_name` - (Optional) The name of the EventHub that Log Analytics Data Export Rule exports too.
```

also given we are giving the id above? what is the point of this? could we not just grab it from the ID and set it ourselves so the user doesn't have to?

Author: Closed this comment as duplicate
